services/web/server/src/simcore_service_webserver/projects/projects_ports_handlers.py

```diff
@@ -0,0 +1,207 @@
+            ) from exc
+
+    return wrapper
+
+
+async def _get_validated_workbench_model(
+    app: web.Application, project_id: ProjectID, user_id: UserID
+) -> dict[NodeID, Node]:
+
+    project: ProjectDict = await projects_api.get_project_for_user(
+        app,
+        project_uuid=f"{project_id}",
+        user_id=user_id,
+        include_templates=False,
+        include_state=False,
+    )
+
+    workbench = parse_obj_as(dict[NodeID, Node], project["workbench"])
+    return workbench
+
+
+#
+# projects/*/inputs COLLECTION -------------------------
+#
+
+routes = web.RouteTableDef()
+
+
+class ProjectPort(BaseModel):
+    key: NodeID = Field(
+        ...,
+        description="Project port's unique identifer. Same as the UUID of the associated port node",
+    )
+    value: Any = Field(..., description="Value assigned to this i/o port")
+
+
+class ProjectPortGet(ProjectPort):
+    label: str
+
+
+@routes.get(f"/{VTAG}/projects/{{project_id}}/inputs", name="get_project_inputs")
+@login_required
+@permission_required("project.read")
+@_handle_project_exceptions
+async def get_project_inputs(request: web.Request) -> web.Response:
+    req_ctx = RequestContext.parse_obj(request)
+    path_params = parse_request_path_parameters_as(ProjectPathParams, request)
+
+    assert request.app  # nosec
+
+    workbench = await _get_validated_workbench_model(
+        app=request.app, project_id=path_params.project_id, user_id=req_ctx.user_id
+    )
+    inputs: dict[NodeID, Any] = _ports.get_project_inputs(workbench)
+
+    return _web_json_response_enveloped(
+        data={
+            node_id: ProjectPortGet(
+                key=node_id, label=workbench[node_id].label, value=value
+            )
+            for node_id, value in inputs.items()
+        }
+    )
+
+
+@routes.patch(f"/{VTAG}/projects/{{project_id}}/inputs", name="update_project_inputs")
+@login_required
+@permission_required("project.update")
+@_handle_project_exceptions
+async def update_project_inputs(request: web.Request) -> web.Response:
+    db: ProjectDBAPI = ProjectDBAPI.get_from_app_context(request.app)
+    req_ctx = RequestContext.parse_obj(request)
+    path_params = parse_request_path_parameters_as(ProjectPathParams, request)
+    inputs_updates = await parse_request_body_as(list[ProjectPort], request)
+
+    assert request.app  # nosec
+
+    workbench = await _get_validated_workbench_model(
+        app=request.app, project_id=path_params.project_id, user_id=req_ctx.user_id
+    )
+    current_inputs: dict[NodeID, Any] = _ports.get_project_inputs(workbench)
+
+    # build workbench patch
+    partial_workbench_data = {}
+    for input_update in inputs_updates:
+        node_id = input_update.key
+        if node_id not in current_inputs.keys():
+            raise web.HTTPBadRequest(reason=f"Invalid input key [{node_id}]")
+
+        # TODO: validate input_update.value against json-schema!
```

Comment:
Should we not avoid leaving these in the code? Is it still valid?

Reply:
missed! Thx